modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/siege/SiegePaymentTask.java

```diff
@@ -19,24 +20,40 @@ public void run() {
         // and now process payments every morning at 1:01 AM, as long as it has
         // been 23 hours after the last payout
         long secsElapsed = (System.currentTimeMillis() - siege.getLastPayout()) / 1000;
+
         if (secsElapsed > 23 * 60 * 60) {
             Calendar rightNow = Calendar.getInstance();
             int hour = rightNow.get(Calendar.HOUR_OF_DAY);
             int minute = rightNow.get(Calendar.MINUTE);
             if ((hour == 1) && (minute == 1)) {
-                siege.setLastPayout(System.currentTimeMillis());
-                //for (String tSiegeName : Settings.SiegeName) {
                 for (World tW : Movecraft.getInstance().getServer().getWorlds()) {
                     ProtectedRegion tRegion = Movecraft.getInstance().getWorldGuardPlugin().getRegionManager(tW).getRegion(siege.getCaptureRegion());
+
                     if (tRegion != null) {
-                        for (String tPlayerName : tRegion.getOwners().getPlayers()) {
-                            int share = siege.getDailyIncome() / tRegion.getOwners().getPlayers().size();
-                            Movecraft.getInstance().getEconomy().depositPlayer(tPlayerName, share);
-                            Movecraft.getInstance().getLogger().log(Level.INFO, String.format(I18nSupport.getInternationalisedString("Siege - Ownership Payout Console"), tPlayerName, share, siege.getName()));
-                        }
+                        payRegion(tRegion);
+                        return;
                     }
                 }
             }
         }
     }
+
+    private void payRegion(ProtectedRegion region) {
+        ArrayList<OfflinePlayer> owners = new ArrayList<>();
```

Comment:
Use a set or some other structure to avoid adding duplicate UUID's in the next two loops